aten/src/TH/generic/THTensorApply.hpp

```diff
@@ -61,9 +61,13 @@ if (std::isnan(val)) break;
 #define th_isnan_break(val)
 #endif
 
-#ifdef _WIN32
-// MSVC doesn't support loop pragmas, but does support others. Create a new macro to account for those differences.
+#ifdef _MSC_VER
+# if _MSC_VER < 1920
+// MSVC < 2019 doesn't support loop pragmas.
 #define PRAGMA_LOOP(P)    // Noop
+# else
+#define PRAGMA_LOOP(P)    __pragma(P)
```

Comment:
If you want to follow the indenting convention here, you should put two spaces after the hash here, so `#  define PRAGMA_LOOP(P) ...`. The idea is that the spaces after the hash are the indentation. I quite like this convention, it makes nested if-statements in preprocessor much more readable.